src/utils/useActivationStatus.ts

```diff
@@ -0,0 +1,33 @@
+import { ApolloError } from "@apollo/client";
+import { useActivationAddressQuery } from "src/generated/graphql";
+import { useLoginSession } from "./useLoginSession";
+
+interface ActivationResult {
+  loading: boolean;
+  error?: ApolloError;
+  activated: boolean;
+}
+
+/**
+ * Queries the activation status of the current account.
+ *
+ * @returns {ActivationResult} A object with two properties: `loading` and `activated`. They are pretty self-explanatory.
+ */
+
+export function useActivationStatus(usePolling?: boolean): ActivationResult {
```

Comment:
I'd prefer assigning default value for this.

Reply:
Since the type of `usePolling` is `boolean | undefined`, the default value is `undefined` when not specified. Also, since `undefined` is a falsy value, it works as a default `false` value.

Reply:
if then, explicitly specify default behavior like `(usePolling: boolean = false)` would be more strict rather than rely on falsy-ness. small difference but still.

Reply:
Resolved in [7e1577d](https://github.com/planetarium/9c-launcher/pull/2110/commits/7e1577d56067ba324483d7b3d5591b85ae10720c)